### PR TITLE
ciscoduo content

### PR DESCRIPTION
**Pull request by spsocprime**
## Proposed Changes

Added analytic content for Cisco Duo data connector

### CODE REVIEW

**Comment by NikTripathi** [comment]
@aprakash13 Please review analytic rules and hunting query. Thanks.

**Comment by v-jayakal** [comment]
@aprakash13  - Please review the analytics, thanks

**Comment by spsocprime** [comment]
We use Parser/Function `CiscoDuo` in queries which was developed for this data connector. Validation fails due to its name

**Comment by v-sabiraj** [comment]
Workbooks Queries are failing. Please look into it.
image
image
image

**Comment by spsocprime** [comment]
> Workbooks Queries are failing. Please look into it. image image image

This is due to usage of old parser
check please parser update PR(https://github.com/Azure/Azure-Sentinel/pull/3512)
There should not be errors in workbook after PR for parser is merged

**Comment by vmanojreddy** [comment]
> > Workbooks Queries are failing. Please look into it. image image image
> 
> This is due to usage of old parser check please parser update PR(#3512) There should not be errors in workbook after PR for parser is merged

**Comment by vmanojreddy** [comment]
> > > Workbooks Queries are failing. Please look into it. image image image
> > 
> > 
> > This is due to usage of old parser check please parser update PR(#3512) There should not be errors in workbook after PR for parser is merged

**Comment by vmanojreddy** [comment]
> > Workbooks Queries are failing. Please look into it. image image image
> 
> This is due to usage of old parser check please parser update PR(#3512) There should not be errors in workbook after PR for parser is merged

After updated the parser also the query is falling.Please look into it. 
image

File: Solutions/CiscoDuoSecurity/Analytic Rules/CiscoDuoADSyncFailed.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+id: 398dd1cd-3251-49d8-b927-5b93bae4a094
+name: Cisco Duo - Admin user deleted
+description: |
+  'Detects when admin user is deleted.'
```
Comment (aprakash13) [inline Solutions/CiscoDuoSecurity/Analytic Rules/CiscoDuoADSyncFailed.yaml:4]: Wondering if the name and description should be AD Sync Failed looking at the logic. It seems there is another analytic rule for Admin Delete.

File: Solutions/CiscoDuoSecurity/Analytic Rules/CiscoDuoAdminMFAFailures.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+triggerThreshold: 0
+tactics:
+  - InitialAccess
+relevantTechniques:
+  - T1078
+query: |
+  let threshold = 10;
+  CiscoDuo
+  | where DvcAction =~ "admin_2fa_error"
+  | summarize count() by DstUserName, bin(TimeGenerated, 10m)
```
Comment (aprakash13) [inline Solutions/CiscoDuoSecurity/Analytic Rules/CiscoDuoAdminMFAFailures.yaml:22]: Might be a good idea to run this query and check. After the summarize statement it seems like the result will have only three columns DstUserName, TimeGenerated and count_. You cannot extend IPCustomEntity and use it in Entity mapping if the column is not there. If there are other columns relevant to the result, please out put them as well when doing summarize.

File: Solutions/CiscoDuoSecurity/Analytic Rules/CiscoDuoMultipleUserLoginFailures.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+relevantTechniques:
+  - T1078
+query: |
+  let threshold = 10;
+  CiscoDuo
+  | where EventType =~ 'authentication'
+  | where EventResult in~ ('denied', 'failure')
+  | summarize count() by DstUserName, bin(TimeGenerated, 10m)
+  | where count_ > threshold
+  | extend IPCustomEntity = SrcIpAddr, AccountCustomEntity = DstUserName
```
Comment (aprakash13) [inline Solutions/CiscoDuoSecurity/Analytic Rules/CiscoDuoMultipleUserLoginFailures.yaml:25]: Might be a good idea to run this query and check. After the summarize statement it seems like the result will have only three columns DstUserName, TimeGenerated and count_. You cannot extend IPCustomEntity and use it in Entity mapping if the column is not there.

File: Solutions/CiscoDuoSecurity/Hunting Queries/CiscoDuoUnpachedAccessDevices.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+requiredDataConnectors:
+  - connectorId: CiscoDuoSecurity
+    dataTypes:
+      - CiscoDuo
+tactics:
+  - InitialAccess
+relevantTechniques:
+  - T1078
+query: |
+  let os_latest = 'x.x.xxx'
```
Comment (aprakash13) [inline Solutions/CiscoDuoSecurity/Hunting Queries/CiscoDuoUnpachedAccessDevices.yaml:15]: Might be a good idea to put a comment that we need to put the latest Os version before running the query.

**Review by aprakash13 — CHANGES_REQUESTED** [review]
Overall, the queries look good. Have added a few suggestions. Please have a look.

**Comment by NikTripathi** [comment]
> > Workbooks Queries are failing. Please look into it. image image image
> 
> This is due to usage of old parser check please parser update PR(#3512) There should not be errors in workbook after PR for parser is merged

After saving that PR still workbook query is failing. Please check. Thanks

**Comment by v-sabiraj** [comment]
Hey @spsocprime, We have resolved the issue for PR Validations failure and request you to please update your PR branch with the latest changes from **Azure:master** so that the PR Validations can be executed.... Thanks!!!

**Comment by spsocprime** [comment]
fixed

**Comment by vmanojreddy** [comment]
@aprakash13 Please review the fixes for analytics. thanks

**Comment by spsocprime** [comment]
> > > Workbooks Queries are failing. Please look into it. image image image
> > 
> > 
> > This is due to usage of old parser check please parser update PR(#3512) There should not be errors in workbook after PR for parser is merged
> 
> After saving that PR still workbook query is failing. Please check. Thanks

image
Worbook queries where fixed. Please refer to the lates version of the workbook.
Analytic rules where fixed

**Review by vmanojreddy — APPROVED** [review]

File: Solutions/CiscoDuoSecurity/Hunting Queries/CiscoDuoUnpachedAccessDevices.yaml (hunk 4 above)
Comment (aprakash13) [reply Solutions/CiscoDuoSecurity/Hunting Queries/CiscoDuoUnpachedAccessDevices.yaml]: Done by the author.

**Comment by aprakash13** [comment]
> @aprakash13 Please review the fixes for analytics. thanks

Two of the 

> @aprakash13 Please review the fixes for analytics. thanks

Thanks @vmanojreddy. 
@spsocprime team - I have marked as resolved the ones that have been fixed. Can you please check the other two and update them as well.

File: Solutions/CiscoDuoSecurity/Analytic Rules/CiscoDuoAdminMFAFailures.yaml (hunk 2 above)
Comment (spsocprime) [reply Solutions/CiscoDuoSecurity/Analytic Rules/CiscoDuoAdminMFAFailures.yaml:22]: fixed

File: Solutions/CiscoDuoSecurity/Analytic Rules/CiscoDuoMultipleUserLoginFailures.yaml (hunk 3 above)
Comment (spsocprime) [reply Solutions/CiscoDuoSecurity/Analytic Rules/CiscoDuoMultipleUserLoginFailures.yaml]: fixed

**Comment by spsocprime** [comment]
fixed

**Review by aprakash13 — APPROVED** [review]
Suggested changes fixed. Approving this now.

**Pull request merged**